tests/jlm/util/TestHashSet.cpp

```diff
@@ -62,6 +62,30 @@ TestUniquePointer()
   assert(hashSet.Size() == 0);
 }
 
+static void
+TestPair()
+{
+  jlm::util::HashSet<std::pair<int, int>> hashSet{ { 1, 10 }, { 5, 50 } };
+
+  // Inserting new value
+  auto result = hashSet.Insert({ 7, 70 });
+  assert(result && hashSet.Size() == 3);
+
+  // Try inserting already inserted
+  result = hashSet.Insert({ 1, 10 });
+  assert(!result && hashSet.Size() == 3);
+
+  // Contains is only true in the correct order
+  assert(hashSet.Contains({ 5, 50 }));
+  assert(!hashSet.Contains({ 50, 5 }));
+
+  // Removing works
+  result = hashSet.Remove({ 5, 50 });
+  assert(result && hashSet.Size() == 2);
+  result = hashSet.Remove({ 5, 50 });
+  assert(!result && hashSet.Size() == 2);
+}
+
```

Comment:
Could you make this its own unit test again, i.e., put a `JLM_REGISTER_UNIT_TEST()` for it instead of adding it in the `Test()` method?

Reply:
Sure, I assume it's fine to do it for all then

Reply:
It absolutely is. Thanks a lot!